src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -171,22 +174,23 @@ public static EventTitle parseEventTitle(String eventTitle) throws ParseExceptio
     }
 
     /**
-     * Parses a {@code String Date} into a {@code Date}. {@code isFutureDateAllowed} is true
-     * if a date after the current date is allowed, false otherwise.
+     * Parses a {@code String Date} into a {@code Date}.
+     * {@code isFutureDateAllowed} is trueif a date after the current date is allowed, false otherwise.
      * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code date} is invalid.
      */
     public static Date parseDate(String date, boolean isFutureDateAllowed) throws ParseException {
         requireNonNull(date);
+        requireNonNull(isFutureDateAllowed);
         String trimmedDate = date.trim();
 
         //Check if date format is valid.
-
         if (!Date.isValidDateFormat(date)) {
             throw new ParseException(Date.MESSAGE_CONSTRAINTS);
         }
 
         //Check if date is after current date and if it is allowed.
-
         if (Date.isAfterCurrentDate(date) & !isFutureDateAllowed) {
```

Comment:
I think this should be a logical AND operation instead of a bitwise AND.

```suggestion
        if (Date.isAfterCurrentDate(date) && !isFutureDateAllowed) {
```